drivers/include/adxl345.h

```diff
@@ -0,0 +1,234 @@
+} adxl345_data_t;
+
+/**
+ * @brief Interrupt configuration struct for the ADXL345 sensor
+ */
+typedef struct {
+    uint8_t source;       /**< Source of interrupts */
+    uint8_t map;          /**< Interrupt mapping control */
+    uint8_t enable;       /**< Interrupt enable control */
+    uint8_t thres_tap;    /**< Tap threshold */
+    uint8_t thres_dur;    /**< Tap duration */
+    uint8_t thres_latent; /**< Tap latency */
+    uint8_t thres_window; /**< Tap window */
+    uint8_t thres_act;    /**< Activity threshold */
+    uint8_t thres_inact;  /**< Inactivity threshold */
+    uint8_t time_inact;   /**< Inactivity time */
+    uint8_t thres_ff;     /**< Free-fall threshold */
+    uint8_t time_ff;      /**< Time threshold */
+    uint8_t act_inact;    /**< Axis enable control for activity and inactivity detection */
+    uint8_t tap_axes;     /**< Axis control for single tap/double tap */
+} adxl345_interrupt_t;
+
+/**
+ * @brief Configuration struct for the ADXL345 sensor
+ */
+typedef struct {
+    i2c_t i2c;                     /**< I2C device which is used */
+    uint8_t addr;                  /**< I2C address */
+    adxl345_interrupt_t interrupt; /**< Interrupts configuration  */
+    uint8_t offset[3];             /**< offset axis */
+    uint8_t range;                 /**< Sensitivity configuration */
+    uint8_t rate;                  /**< Configured sample rate for accel  */
+    uint8_t full_res;              /** < Resolution bit */
+    uint8_t scale_factor;          /** < Scale factor for converting value to mg */
+} adxl345_params_t;
+
+/**
+ * @brief Device descriptor for the ADXL345 sensor
+ */
+typedef struct {
+    adxl345_params_t params;      /**< Device configuration */
+} adxl345_t;
+
+/**
+ * @brief   Initialize the ADXL345 accelerometer driver.
+ *
+ * @param[out] dev          device descriptor of accelerometer to initialize
+ * @param[in]  params       configuration parameters
+ *
+ * @return                  ADXL345_OK on success
+ * @return                  ADXL345_NOI2C if initialization of I2C bus failed
+ * @return                  ADXL345_NODEV if accelerometer test failed
+ */
+int adxl345_init(adxl345_t *dev,const adxl345_params_t* params);
+
+/**
+ * @brief   Read accelerometer's data
+ *
+ * Acceleration will be calculated as:<br>
+ *     \f$ accel = {value \times 3.9} \f$ if full scale is set to 2g<br>
+ *     \f$ accel = {value \times 7.8} \f$ if full scale is set to 4g<br>
+ *     \f$ accel = {value \times 15.6} \f$ if full scale is set to 8g<br>
+ *     \f$ accel = {value \times 31.2} \f$ if full scale is set to 16g<br>
+ *
+ * @param[in]  dev          device descriptor of accelerometer
+ * @param[out] data         the current acceleration data [in mg]
+ */
+void adxl345_read(adxl345_t *dev, adxl345_data_t *data);
+
+/**
+ * @brief   set ADXL345's interrupts configuration
+ *
+ * @param[in]  dev          device descriptor of accelerometer
+ */
+void adxl345_interrupt(adxl345_t *dev);
```

Comment:
except if you want to expose them. In this case, I would prefer a name like adxl1345_set_xxx.